File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefBooleanProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import org.jetbrains.annotations.NotNull;

public class CTPrefBooleanProperty extends CTPrefProperty<Boolean> {
    public CTPrefBooleanProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, false);
    }

    public CTPrefBooleanProperty(@NotNull CTPreferencesProvider provider, String name, boolean def) {
        super(Boolean.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefBooleanProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider

class CTPrefBooleanProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: Boolean = false
) : CTPrefProperty<Boolean>(
    Boolean::class, provider, name, def
)
